data-prepper-pipeline-parser/src/main/java/org/opensearch/dataprepper/pipeline/parser/rule/RuleEvaluator.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.dataprepper.pipeline.parser.rule;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.jayway.jsonpath.Configuration;
+import com.jayway.jsonpath.JsonPath;
+import com.jayway.jsonpath.Option;
+import com.jayway.jsonpath.ParseContext;
+import com.jayway.jsonpath.PathNotFoundException;
+import com.jayway.jsonpath.ReadContext;
+import com.jayway.jsonpath.spi.json.JacksonJsonProvider;
+import com.jayway.jsonpath.spi.mapper.JacksonMappingProvider;
+import org.opensearch.dataprepper.model.configuration.PipelineModel;
+import org.opensearch.dataprepper.model.configuration.PipelinesDataFlowModel;
+import org.opensearch.dataprepper.pipeline.parser.transformer.PipelineTemplateModel;
+import org.opensearch.dataprepper.pipeline.parser.transformer.TransformersFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+public class RuleEvaluator {
+
+    private static final Logger LOG = LoggerFactory.getLogger(RuleEvaluator.class);
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final ObjectMapper yamlMapper = new ObjectMapper(new YAMLFactory());
+    private final TransformersFactory transformersFactory;
+    private String PLUGIN_NAME = null;
+
+    public RuleEvaluator(TransformersFactory transformersFactory) {
+        this.transformersFactory = transformersFactory;
+    }
+
+    public RuleEvaluatorResult isTransformationNeeded(PipelinesDataFlowModel pipelineModel) {
+        return isDocDBSource(pipelineModel);
+    }
+
+    /**
+     * Evaluates model based on pre defined rules and
+     * result contains the name of the pipeline that will need transformation,
+     * evaluated boolean result and the corresponding template model
+     * Assumption: only one pipeline can have transformation.
+     * @param pipelinesModel
+     * @return RuleEvaluatorResult
+     */
+    private RuleEvaluatorResult isDocDBSource(PipelinesDataFlowModel pipelinesModel) {
+        PLUGIN_NAME = "documentdb";
```

Comment:
This is only a temporary implementation, will move towards scanning folder and fetching and applying rule file dynamically.